examples_tests/50.MeshPacking/main.cpp

```diff
@@ -0,0 +1,388 @@
+
+    vec3 normal = texelFetch(MeshPackedData_A2B10G10R10_SNORM_PACK32,int(gl_VertexIndex)+vertexNormal.dataOffsetTable[gl_DrawID]).xyz;
+    //Normal = normalMat*normalize(vNormal);
+    Normal = vec3(1.0, 0.0, 0.0);
+}
+)"
+
+}
+;
+
+core::smart_refctd_ptr<asset::ICPUSpecializedShader> createModifiedVertexShader(asset::ICPUSpecializedShader* _fs)
+{
+    const asset::ICPUShader* unspec = _fs->getUnspecialized();
+    assert(unspec->containsGLSL());
+
+    auto begin = reinterpret_cast<const char*>(unspec->getSPVorGLSL()->getPointer());
+    auto end = begin + unspec->getSPVorGLSL()->getSize();
+    std::string resultShaderSrc(begin, end);
+
+    size_t firstNewlineAfterVersion = resultShaderSrc.find("\n", resultShaderSrc.find("#version "));
+
+    const std::string buffersDef = SHADER_OVERRIDES[0];
+    const std::string mainDef = SHADER_OVERRIDES[1];
+
+    resultShaderSrc.insert(firstNewlineAfterVersion, buffersDef);
+    resultShaderSrc += mainDef;
+
+    auto unspecNew = core::make_smart_refctd_ptr<asset::ICPUShader>(resultShaderSrc.c_str());
+    auto specinfo = _fs->getSpecializationInfo();
+    auto vsNew = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(unspecNew), std::move(specinfo));
+
+    return vsNew;
+}
+
+struct DrawIndexedIndirectInput
+{
+    asset::SBufferBinding<video::IGPUBuffer> vtxBindings[video::IGPUMeshBuffer::MAX_ATTR_BUF_BINDING_COUNT];
+    asset::E_PRIMITIVE_TOPOLOGY mode = asset::EPT_TRIANGLE_LIST;
+    asset::E_INDEX_TYPE indexType = asset::EIT_16BIT;
+    core::smart_refctd_ptr<video::IGPUBuffer> indexBuff = nullptr;
+    core::smart_refctd_ptr<video::IGPUBuffer> indirectDrawBuff = nullptr;
+    size_t offset = 0u;
+    size_t maxCount = 0u;
+    size_t stride = 0u;
+    core::smart_refctd_ptr<video::IGPUBuffer> countBuffer = nullptr;
+    size_t countOffset = 0u;
+};
+
+struct DrawIndexedIndirectInputV2
+{
+    asset::SBufferBinding<video::IGPUBuffer> vtxBuffer;
+    asset::E_PRIMITIVE_TOPOLOGY mode = asset::EPT_TRIANGLE_LIST;
+    asset::E_INDEX_TYPE indexType = asset::EIT_16BIT;
+    core::smart_refctd_ptr<video::IGPUBuffer> idxBuffer = nullptr;
+    core::smart_refctd_ptr<video::IGPUBuffer> indirectDrawBuff = nullptr;
+    size_t offset = 0u;
+    size_t maxCount = 0u;
+    size_t stride = 0u;
+    core::smart_refctd_ptr<video::IGPUBuffer> countBuffer = nullptr;
+    size_t countOffset = 0u;
+};
+
+void packMeshBuffers(video::IVideoDriver* driver, core::vector<ICPUMeshBuffer*>& meshBuffers, SVertexInputParams& vipOutput, DrawIndexedIndirectInput& output)
+{
+    using MeshPacker = CCPUMeshPackerV1<DrawElementsIndirectCommand_t>;
+
+    MeshPacker::PackerDataStore packedMeshBuffer;
+    //core::smart_refctd_ptr<IGPUBuffer> gpuIndirectDrawBuffer;
+
+    MeshPacker::AllocationParams allocParams;
+    allocParams.indexBuffSupportedCnt = 20000000u;
+    allocParams.indexBufferMinAllocSize = 5000u;
```

Comment:
what's the status on the bug when you set it to `1u` ?

>packing sponza this way works incorrectly (it is all good if I change #1 and #2 to 5000u),

Reply:
I haven't fixed that yet